File: CS/RangeDataSource/DataSourceHelper.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;

namespace RangeDataSource {
    public static class DataSourceHelper {
        public static bool IsFieldExists(object dataSource, string fieldName, Type reqType) {
            PropertyDescriptorCollection pdc = ((ITypedList)dataSource).GetItemProperties(null);
            foreach (PropertyDescriptor pd in pdc) {
                if (pd.DisplayName == fieldName) {
                    if (reqType == null || pd.PropertyType == reqType) return true;
                }
            }
            return false;
        }
    }
}

File: VB/RangeDataSource/DataSourceHelper.vb
Imports System
Imports System.ComponentModel

Namespace RangeDataSource

    Public Module DataSourceHelper

        Public Function IsFieldExists(ByVal dataSource As Object, ByVal fieldName As String, ByVal reqType As Type) As Boolean
            Dim pdc As PropertyDescriptorCollection = CType(dataSource, ITypedList).GetItemProperties(Nothing)
            For Each pd As PropertyDescriptor In pdc
                If Equals(pd.DisplayName, fieldName) Then
                    If reqType Is Nothing OrElse pd.PropertyType Is reqType Then Return True
                End If
            Next

            Return False
        End Function
    End Module
End Namespace
